src/App.tsx

```diff
@@ -14,11 +14,103 @@ const goodsFromServer: string[] = [
   'Garlic',
 ];
 
-const App: React.FC = () => (
-  <div className="App">
-    <h1>Selected good: -</h1>
-    {goodsFromServer.length}
-  </div>
-);
+type State = {
+  selectedGood: string[];
+};
+
+const showGood = (value:string[]) => {
+  let rezult = '';
+
+  if (value.length === 0) {
+    rezult = 'No goods selected';
+  } else if (value.length === 1) {
+    rezult = `${value.join(',')} is selected`;
+  } else {
+    const arr = [...value];
+    const last = arr.pop();
+
+    rezult = `${arr.join(',')} and ${last} are selected`;
+  }
+
+  return rezult;
+};
+
+class App extends React.Component<{}, State> {
+  state = {
+    selectedGood: ['Jam'],
+  };
+
+  addGoods = (item: string) => {
+    this.setState((prev) => {
+      const newSelectedGoods = prev.selectedGood;
+
+      newSelectedGoods.push(item);
```

Comment:
You didn't make copy here
```suggestion
      const newSelectedGoods = [...prev.selectedGood];

      newSelectedGoods.push(item);
```